lib/fluent/plugin/output.rb

```diff
@@ -326,6 +326,7 @@ def start
           @buffering = prefer_buffered_processing
           if !@buffering && @buffer
             @buffer.terminate # it's not started, so terminate will be enough
+            @buffer = nil
```

Comment:
I understand the situation. Adding comment is better with "Avoid side-effect that in_monitor_agent shows non-buffered plugins as buffered."